src/Simulation/Common/QubitManager.cs

```diff
@@ -25,11 +26,13 @@ public class QubitManager
         long AllocatedForBorrowing; // All qubits allocated only for borrowing, will be marked with this number or higher.
         long[] qubits; // Tracks the allocation state of all qubits.
         long free; // Points to the first free (unallocated) qubit.
+        long freeTail; // Points to the last free (unallocated) qubit. Only valid iff (!EncourageReuse).
```

Comment:
For my education: why do we need dynamic Allocated/Disabled markers rather than using fixed sentinel values (to avoid remarking the qubits when the manager is extended)?

Reply:
I'd also suggest renaming `free` to `freeHead` to match `freeTail` (and maybe expanding the comment, explaining that this is a singly-linked free list).